src/util/Timer.cpp

```diff
@@ -294,7 +298,8 @@ VirtualClock::advanceExecutionQueue()
         mExecutionIterator = mExecutionQueue.begin();
     }
 
-    YieldTimer queueYt(*this, MAX_EXECUTION_QUEUE_DURATION,
+    YieldTimer queueYt(*this, *mExecutionDebtTracker,
```

Comment:
Something is off here:
intuitively we should always process some SCP messages regardless of how the node is doing (what's the point of the node just picking up stuff from the connections if it doesn't do anything with it?), and right now if we close a ledger, we're going to potentially not process anything SCP related.

Maybe we should just add new "types" of execution so that can get both strategies. We end up with:
* NORMAL_EVENT, anything needed for the node to contribute to SCP, p2p control messages
* LOW_PRIORITY_EVENT, WorkScheduler
* DROPPABLE_EVENT, ultra low priority - tx flooding for example

Switch to a pure priority order:
only pick up items from a lower priority queue if there are no pending items from a higher priority. Where NORMAL_EVENT > LOW_PRIORITY_EVENT > DROPPABLE_EVENT.

We can then have different settings for each of those, something like:
* max queue size before we trigger some sort of recovery strategy (similar to what I wrote, but the strategy is per event type). We would a low limit for NORMAL (like 100 events).
* how long we're willing to not process items of the given type (ie, for SCP that time is 0: we always want to process items from that queue).
* maybe alternate max numbers when we run out of credits. For example, pick up "up to 1000 items/100 ms" becomes pick up "up to 10 items/10ms" when we don't have credits.

We'd configure:
* NORMAL_EVENT pretty much always process something (maybe slow down a bit if we've overwhelmed the system)
* LOW_PRIORITY_EVENT would execute no work at all for periods of time if it was responsible for hangs
* DROPPABLE_EVENT same than LOW_PRIORITY_EVENT. Would probably have higher tolerance for larger queues as those are "bursty" in nature and cheap to process (so we can recover without dropping even when there are large queues).